src/dune_rules/merlin/merlin.ml

```diff
@@ -466,12 +466,20 @@ module Unprocessed = struct
        } as t) sctx ~dir ~more_src_dirs ~expander =
     let open Action_builder.O in
     let+ config =
-      let* stdlib_dir =
+      let* stdlib_dir, extra_obj_dirs =
         Action_builder.of_memo
         @@
         match t.config.mode with
-        | `Ocaml -> Memo.return (Some stdlib_dir)
-        | `Melange -> Melange_binary.where sctx ~loc:None ~dir
+        | `Ocaml -> Memo.return (Some stdlib_dir, [])
+        | `Melange -> (
+          let open Memo.O in
+          let+ dirs = Melange_binary.where sctx ~loc:None ~dir in
+          match dirs with
+          | [] -> (None, [])
+          | [ stdlib_dir ] -> (Some (Path.external_ stdlib_dir), [])
```

Comment:
If we are converting to `Path.t` here using `external_` in this line and 482 below, would it make sense to just return `Path.t list Memo.t` from `Melange_binary.where` directly?

Reply:
See https://github.com/ocaml/dune/pull/7176#discussion_r1118123679

Reply:
Actually, `Bin.parse_sep` doesn't allow external paths either.

Reply:
I changed it to use `Bin.parse_sep`.